src/core/connection.h

```diff
@@ -8,27 +8,36 @@ namespace mavsdk {
 
 class Connection {
 public:
-    typedef std::function<void(mavlink_message_t& message)> receiver_callback_t;
+    typedef std::function<void(mavlink_message_t& message, Connection* connection)>
+        receiver_callback_t;
 
-    Connection(receiver_callback_t receiver_callback);
+    Connection(
+        receiver_callback_t receiver_callback,
+        ForwardingOption forwarding_option = ForwardingOption::ForwardingOff);
     virtual ~Connection();
 
     virtual ConnectionResult start() = 0;
     virtual ConnectionResult stop() = 0;
 
     virtual bool send_message(const mavlink_message_t& message) = 0;
 
+    bool do_forward_messages() const;
+    uint8_t forwarding_connections_count() const;
+
     // Non-copyable
     Connection(const Connection&) = delete;
     const Connection& operator=(const Connection&) = delete;
 
 protected:
     bool start_mavlink_receiver();
     void stop_mavlink_receiver();
-    void receive_message(mavlink_message_t& message);
+    void receive_message(mavlink_message_t& message, Connection* connection);
 
     receiver_callback_t _receiver_callback{};
     std::unique_ptr<MAVLinkReceiver> _mavlink_receiver;
+    ForwardingOption _forwarding_option;
+
+    static uint8_t _forwarding_connections_count;
```

Comment:
I think this can be a `unsigned`, sorry for confusing you earlier.

I also wonder whether this should be a `static std::atomic<unsigned>` because a new connection is added in different thread than where the receiver thread.

Reply:
Yes, you are right, I missed that !